From d8bef562f1959ab5b0e8b3dbc7d263326bf708b2 Mon Sep 17 00:00:00 2001
From: Yuri Tolstov <ytolstov@marvell.com>
Date: Fri, 23 Jul 2021 06:53:51 -0700
Subject: [PATCH 13/13] octeontx2-af: Reduce TIM clock divider values

commit 69fafd2b7e943bfd4c9ecd7b4fe366ed022595de from
git@git.assembla.com:cavium/WindRiver.linux.git

Reduced TIM clock divider values for GTI, TENNS, and PTP clocks
to increase timer resolutions.

Signed-off-by: Yuri Tolstov <ytolstov@marvell.com>
Change-Id: I44564f628f69543490498f2214710f130dbdbc6f
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/57545
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Jerin Jacob Kollanukkaran <jerinj@marvell.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_tim.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_tim.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_tim.c
index ed6143e8c2f4..eb84172bb119 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_tim.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_tim.c
@@ -161,14 +161,14 @@ int rvu_mbox_handler_tim_config_ring(struct rvu *rvu,
 
 	switch (req->clocksource) {
 	case TIM_CLK_SRCS_TENNS:
-		intervalmin = 256;
+		intervalmin = 200;
 		break;
 	case TIM_CLK_SRCS_GPIO:
 		intervalmin = 256;
 		break;
 	case TIM_CLK_SRCS_GTI:
 	case TIM_CLK_SRCS_PTP:
-		intervalmin = 300;
+		intervalmin = 200;
 		break;
 	default:
 		return TIM_AF_INVALID_CLOCK_SOURCE;
-- 
2.31.1

